components/support/sync-telemetry/src/main/java/mozilla/components/support/sync/telemetry/SyncTelemetry.kt

```diff
@@ -284,4 +287,42 @@ object SyncTelemetry {
         val message = reason.message ?: "Unexpected error: ${reason.code}"
         metric.set(message.take(MAX_FAILURE_REASON_LENGTH))
     }
+
+    fun processFxaTelemetry(jsonStr: String) {
+        val json = try {
+            JSONObject(jsonStr)
+        } catch (e: JSONException) {
+            logger.error("Invalid JSON in FxA telemetry", e)
```

Comment:
(throughout this function)

Do you care to know if this actually happens in the wild?

Three options:
- crash the app here by throwing
- submit a caught exception to Sentry (via CrashReporter, which you'd need to pass-into this function)
- set some "faulty fxa telemetry detected" glean boolean metric

I think we should do at least one of these (and probably not crash), unless there's some other way to verify that telemetry is broken.